internal/api/mail.go

```diff
@@ -396,22 +441,28 @@ func (a *API) sendEmailChange(tx *storage.Connection, config *conf.GlobalConfigu
 	}
 
 	u.EmailChangeSentAt = &now
-	return errors.Wrap(tx.UpdateOnly(
+	err = tx.UpdateOnly(
 		u,
 		"email_change_token_current",
 		"email_change_token_new",
 		"email_change",
 		"email_change_sent_at",
 		"email_change_confirm_status",
-	), "Database error updating user for email change")
+	)
+
+	if err != nil {
+		return errors.Wrap(err, "Database error updating user for email change")
+	}
+
+	return nil
 }
 
 func validateEmail(email string) (string, error) {
 	if email == "" {
-		return "", unprocessableEntityError("An email address is required")
+		return "", badRequestError(ErrorCodeValidationFailed, "An email address is required")
```

Comment:
would this be considered a breaking change since we are changing the error code returned?

Reply:
It's not a breaking change, as there was no official contract for the status codes. Some of the status codes are also incorrect (use of 401 instead of 403).